File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--updating-the-controls-on-a-ribbon-at-run-time_4.vb
    Private Sub CustomerRibbon_Load(ByVal sender As System.Object, _
   ByVal e As Microsoft.Office.Tools.Ribbon.RibbonUIEventArgs) _
       Handles MyBase.Load

        customerTable = nwDataSet.Customers
        customersTableAdapter.Fill(customerTable)

        Dim customerQuery = From customers In customerTable.AsEnumerable.Take(20) _
                Select CustomerID = customers.Customer_ID, _
                CustomerName = customers.Contact_Name

        ' Execute the query.
        For Each item In customerQuery
            Me.ComboBox1.Items.Add(CreateRibbonDropDownItem())

            Me.ComboBox1.Items.Last().Label = item.CustomerID.ToString() _
                + "|" + item.CustomerName
        Next item

        Me.ComboBox1.Text = Me.ComboBox1.Items.First().Label
        PopulateSalesOrderInfo()
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--updating-the-controls-on-a-ribbon-at-run-time_4.cs
        private void CustomerRibbon_Load(object sender, RibbonUIEventArgs e)
        {
            customerTable = nwDataSet.Customers;
            customerTableAdapter.Fill(customerTable);

            var customerQuery = from customers in customerTable.AsEnumerable().Take(20)
                                select new
                                {
                                    CustomerID = customers.Field<string>("Customer ID"),
                                    CustomerName = customers.Field<string>("Contact Name")
                                };


            // Execute the query.
            foreach (var item in customerQuery)
            {
                this.comboBox1.Items.Add(CreateRibbonDropDownItem());
                this.comboBox1.Items.Last().Label =
                item.CustomerName + "|" + item.CustomerID.ToString();
            }
            this.comboBox1.Text = this.comboBox1.Items.First().Label;
            PopulateSalesOrderInfo();
        }